diesel/src/mysql/connection/bind.rs

```diff
@@ -40,6 +71,56 @@ impl Binds {
         &mut self.mysql_binds
     }
 
+    /// The caller of this function must immediately check for errors
+    /// after return
+    pub fn populate_dynamic_buffers(&mut self, stmt: *mut ffi::MYSQL_STMT) {
+        use std::cmp::min;
+
+        for (i, buffer) in self.data.iter_mut().enumerate() {
+            if self.errors.as_ref().unwrap()[i] != 0 {
+                let written_bytes = buffer.len();
+                let needed_capacity = self.lengths[i] as usize;
+                buffer.reserve(needed_capacity - written_bytes);
+                self.mysql_binds[i].buffer = buffer.as_mut_ptr() as *mut libc::c_void;
```

Comment:
Was true before this as well. This is needed because the call to reserve above is invalidating the previous pointer (Forgetting this line was the cause of the segfault I mentioned in gitter)